addon/services/hover-intent.js

```diff
@@ -0,0 +1,126 @@
+import Ember from 'ember';
+import get from 'ember-metal/get';
+import set from 'ember-metal/set';
+import guidFor from 'ember-metal/utils';
+import $ from 'jquery';
+
+const { assert, Service, computed } = Ember;
+
+/**
+ * This service sets up a single event listener bound to the document for the
+ * 'mousemove' event, and keeps track of the velocity of the mouse. The service
+ * can be given a series of targets (DOM elements) to keep track of, and, if the
+ * mouse's velocity is below a specified threshold, it will loop through the
+ * targets and see if any are hovered. If so, it will invoke a callback
+ * function on that target.
+ *
+ * @class HoverIntent
+ * @extends Ember.Service
+ */
+export default Service.extend({
+  // private properties used to calculate the mouse velocity
+  _mouseX: null,
+  _mouseY: null,
+  _timestamp: null,
+  _mouseVelocity: null,
+  threshold: .1, // pixels moved per millisecond
+
+  mouseVelocity: computed({
+    get() {
+      return this._mouseVelocity;
```

Comment:
This getter can be left as:

```js
return 0;
```

Since it will never be called after the initial `get`